opal/class/opal_pointer_array.c

```diff
@@ -67,6 +70,103 @@ static void opal_pointer_array_destruct(opal_pointer_array_t *array)
+        __b_pos = 0;                                                    \
+                                                                        \
+        if( 0x00000000FFFFFFFFULL == (__check_value & 0x00000000FFFFFFFFULL) ) { \
+            __check_value >>= 32; __b_pos += 32;                        \
+        }                                                               \
+        if( 0x000000000000FFFFULL == (__check_value & 0x000000000000FFFFULL) ) { \
+            __check_value >>= 16; __b_pos += 16;                        \
+        }                                                               \
+        if( 0x00000000000000FFULL == (__check_value & 0x00000000000000FFULL) ) { \
+            __check_value >>= 8; __b_pos += 8;                          \
+        }                                                               \
+        if( 0x000000000000000FULL == (__check_value & 0x000000000000000FULL) ) { \
+            __check_value >>= 4; __b_pos += 4;                          \
+        }                                                               \
+        if( 0x0000000000000003ULL == (__check_value & 0x0000000000000003ULL) ) { \
+            __check_value >>= 2; __b_pos += 2;                          \
+        }                                                               \
+        if( 0x0000000000000001ULL == (__check_value & 0x0000000000000001ULL) ) { \
+            __b_pos += 1;                                               \
+        }                                                               \
+        if( (SET) ) {                                                   \
+            table->free_bits[__b_idx] |= (1ULL << __b_pos);             \
+        }                                                               \
+        (STORE) = (__b_idx * 8 * sizeof(uint64_t)) + __b_pos;           \
+    } while(0)
+
+/**
+ * Set the IDX bit in the free_bits array. The bit should be previously unset.
+ */
+#define SET_BIT(IDX)                                                    \
+    do {                                                                \
+        uint32_t __b_idx, __b_pos;                                      \
+        GET_BIT_POS((IDX), __b_idx, __b_pos);                           \
+        assert( 0 == (table->free_bits[__b_idx] & (1UL << __b_pos)));   \
+        table->free_bits[__b_idx] |= (1ULL << __b_pos);                 \
+    } while(0)
+
+/**
+ * Unset the IDX bit in the free_bits array. The bit should be previously set.
+ */
+#define UNSET_BIT(IDX)                                                  \
+    do {                                                                \
+        uint32_t __b_idx, __b_pos;                                      \
+        GET_BIT_POS((IDX), __b_idx, __b_pos);                           \
+        assert( (table->free_bits[__b_idx] & (1UL << __b_pos)));        \
+        table->free_bits[__b_idx] ^= (1ULL << __b_pos);                 \
+    } while(0)
+
+#if 0
```

Comment:
That function is too expensive to use in general (even in debug mode). But, in case something goes wrong, it is useful to debug.